pkg/ddl/add_column.go

```diff
@@ -172,6 +179,19 @@ func checkAndCreateNewColumn(ctx sessionctx.Context, ti ast.Ident, schema *model
 	return CreateNewColumn(ctx, schema, spec, t, specNewColumn)
 }
 
+func checkUnsupportedCharsetForTiFlash(tbl *model.TableInfo, colName ast.CIStr, colTp types.FieldType) error {
+	colCharset := colTp.GetCharset()
+	if tbl.Charset == charset.CharsetGBK || tbl.Charset == charset.CharsetGB18030 {
+		colCharset = tbl.Charset
+	}
```

Comment:
It is better to use `TiFlashSupportedCharsets` instead of filtering unsupported charsets.